src/full-join.js

```diff
@@ -31,6 +28,10 @@ function fullJoin(
 
       return isMatch;
     });
+    if (options.as) {
+      cur[options.as] = matches;
+      return prev.concat(cur);
```

Comment:
To join without merging objects, similar to 'mongodb $lookup (aggregation)'
So mappers are not required

Reply:
I can't agree with that. If a consumer of the library has an option to put name mappers, they would expect it to work - even if it might not be useful in this particular case.